src/lib/storage/segment_iterables/any_segment_iterable.cpp

```diff
@@ -11,10 +11,12 @@ AnySegmentIterable<T> CreateAnySegmentIterable<T>::create(const BaseSegment& bas
   auto any_segment_iterable = std::optional<AnySegmentIterable<T>>{};
 
   resolve_segment_type<T>(base_segment, [&](const auto& segment) {
-    const auto actual_iterable = create_iterable_from_segment<T>(segment);
+    const auto actual_iterable = create_iterable_from_segment<T, false>(segment);
     any_segment_iterable.emplace(erase_type_from_iterable(actual_iterable));
   });
 
+  DebugAssert(any_segment_iterable, "Could not create iterable");
```

Comment:
I think we rely on resolve_segment_type calling the functor exactly once all over the code base and don't check whether it did anywhere else. I'd remove this assert